src/main/equationServlet.java

```diff
@@ -0,0 +1,120 @@
+package ru.itmo.SergeyPesegov.web.servlets;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.*;
+
+@WebServlet(
+        urlPatterns = {"/calc"}
+)
+public class equationServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+
+        String equation = req.getParameter("equation");
+
+        Map<String, Integer> args = new HashMap<>();
+        Map<String, String> StringValue = new HashMap<>();
+        for (Map.Entry<String, String[]> par : req.getParameterMap().entrySet()) {
+            if (par.getKey().equals("equation")) continue;
+            try {
+                args.put(par.getKey(), Integer.parseInt(par.getValue()[0]));
+            } catch (NumberFormatException e) {
+                StringValue.put(par.getKey(), par.getValue()[0]);
+            }
+        }
+        for (String key : StringValue.keySet()) {
+            Integer value = args.get(StringValue.get(key));
+            args.put(key, value);
+        }
+        List<String> polishTokens = parseIntoPolishNotation(equation);
+        int result = calculate(polishTokens, args);
+        resp.getOutputStream().print(result);
+        resp.getOutputStream().flush();
+        resp.getOutputStream().close();
+    }
+
+    private List<String> parseIntoPolishNotation(String equation) {
+        List<String> tokens = new ArrayList<>();
+        StringTokenizer tokenizer = new StringTokenizer(equation.replaceAll(" ", ""), "()+-/*", true);
+        while (tokenizer.hasMoreElements()) {
+            tokens.add(tokenizer.nextToken());
+        }
+        Deque<String> stack = new ArrayDeque<>();
+        List<String> polishNotation = new ArrayList<>();
+        for (String token : tokens) {
+            try {
+                switch (token) {
+                    case "+":
+                    case "-":
+                    case ")":
+                        while ("+-/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        if (token.equals(")")) {

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4302155)